internal/cri/server/podsandbox/sandbox_run.go

```diff
@@ -321,3 +321,12 @@ func (c *Controller) ensureImageExists(ctx context.Context, ref string, config *
 	}
 	return &newImage, nil
 }
+
+func (c *Controller) getSandboxImageName() string {
+	// SandboxImageNameKey is the name of the sandbox image used to scope pod shared resources used by the pod's containers,
+	// if nil the image service is not being used.
+	if c.imageService != nil {
+		return c.imageService.PinnedImage("sandbox")
```

Comment:
noting over in the other use of sandbox... https://github.com/ChengyuZhu6/containerd/blob/main/internal/cri/server/podsandbox/sandbox_run.go#L78-L81 we have a check for if we have a pinned sandbox name or else we use the default..
```
        sandboxImage := c.imageService.PinnedImage("sandbox")
	if sandboxImage == "" {
		sandboxImage = criconfig.DefaultSandboxImage
	}
```

Should this api return ^ or nil ?

Reply:
Ah, thanks for bring it up. I have moved the logic to `getSandboxImageName` and refactored the code to get the sandbox name.

Reply:
>Should this api return ^ or nil ?

Instead of returning nil, it’s best to return the default sandbox image name.